dcos/servicemanager.py

```diff
@@ -39,4 +45,34 @@ def start_service(self, package_name, package_version, options):
             json['packageVersion'] = package_version
         if options is not None:
             json['options'] = options
-        return self.cosmos.call_endpoint(endpoint, json=json)
+        try:
+            return self.cosmos.call_endpoint(endpoint, json=json)
+        except DCOSAuthenticationException:
+            raise
+        except DCOSAuthorizationException:
+            raise
+        except DCOSHTTPException as e:
+            if e.status() == 404:
+                message = 'Your version of DC/OS ' \
+                          'does not support this operation'
+                raise DCOSException(message)
+            else:
+                return e.response
+
+
+def _get_default_base_url():
+    """
+    Gets the cosmos url
+
+    :returns: cosmos base url
+    :rtype: str
+    """
+    toml_config = config.get_config()
+    base_url = config.get_config_val('package.cosmos_url', toml_config)
+    if base_url is None:
+        base_url = config.get_config_val('core.dcos_url', toml_config)
+        if base_url is None:
+            raise config.missing_config_exception(['core.dcos_url'])
+        else:
+            base_url = urllib.parse.urljoin(base_url, 'cosmos/')
```

Comment:
Ah okay. Then it doesn't make sense to merge this method with the very similar one in `dcos/cosmos.py`, since we have to append `cosmos/` here. Thanks!